src/metabase/analyze/fingerprint/fingerprinters.clj

```diff
@@ -63,25 +66,37 @@
                                       ~v
                                       (catch Throwable _#))]))))
 
-(defmacro ^:private with-reduced-error
-  [msg & body]
-  `(let [result# (sync-util/with-error-handling ~msg ~@body)]
-     (if (instance? Throwable result#)
-       (reduced result#)
-       result#)))
-
 (defn with-error-handling
   "Wrap `rf` in an error-catching transducer."
   [rf msg]
+  ;; This function is written in a specific way to ensure that try-catch blocks don't produce closures.
```

Comment:
Not yet. This is controlled by the Clojure compiler, and most of the time it is reasonable. If you return the a value from both the try and the catch block as the last value of the function, it can generate a simple try/catch with `areturn`. But if you want to perform some action on top of the return value, it has to wrap the try/catch in a closure (this is somewhat similar to TCO).